Diffusion/Common/ImportModelBehavior.swift

```diff
@@ -0,0 +1,97 @@
+        content
+            .fileImporter(isPresented: $importPanelState, allowedContentTypes: [.folder, .zip]) { result in
+                do {
+                    let url = try result.get()
+                    if url.hasDirectoryPath {
+                        let fileManager = FileManager.default
+                        let contents = try fileManager.contentsOfDirectory(atPath: url.path)
+                        
+                        if contents.contains(where: { $0.caseInsensitiveCompare("merges.txt") == .orderedSame }) && contents.contains(where: { $0.caseInsensitiveCompare("vocab.json") == .orderedSame }) {
+                            // Folder contains the required filenames
+                            do {
+                                try fileManager.moveItem(at: url, to: Settings.shared.applicationSupportURL().appendingPathComponent("hf-diffusion-models").appendingPathComponent(url.lastPathComponent))
+                            } catch {
+                                // Error handling
+                                print("Error: \(error.localizedDescription)")
+                            }
+                        } else {
+                            // Folder does not contain the required filenames
+                            isBadSelectionAlertShown = true
+                        }
+                    } else if url.pathExtension == "zip" {
+                        DispatchQueue.global(qos: .background).async {
+                            do {
+                                let extractedURL = Settings.shared.applicationSupportURL().appendingPathComponent("hf-diffusion-models").appendingPathComponent(url.deletingPathExtension().lastPathComponent)
+                                try FileManager.default.createDirectory(at: extractedURL, withIntermediateDirectories: true, attributes: nil)
+
+                                if let archive = Archive(url: url, accessMode: .read) {
+                                    for entry in archive {
+                                        // Skip hidden and __MACOSX files
+                                        let filename = entry.path.components(separatedBy: "/").last ?? ""
+                                        if !filename.hasPrefix(".") && filename.lowercased() != "__macosx" {
+                                            let destinationURL = extractedURL.appendingPathComponent(entry.path)
+                                            let _ = try archive.extract(entry, to: destinationURL)
+                                        }
+                                    }
+                                }
+
+                                let extractedContents = try FileManager.default.contentsOfDirectory(atPath: extractedURL.path)
+                                if extractedContents.contains(where: { $0.caseInsensitiveCompare("merges.txt") == .orderedSame }) && extractedContents.contains(where: { $0.caseInsensitiveCompare("vocab.json") == .orderedSame }) {
```

Comment:
There's a similar check above, maybe factor it out? I wonder if it'd make sense to also check for `.mlpackage` bundles, what do you think?